go/vt/vttablet/tabletserver/query_executor.go

```diff
@@ -1227,10 +1227,63 @@ func (qre *QueryExecutor) recordUserQuery(queryType string, duration int64) {
 	qre.tsv.Stats().UserTableQueryTimesNs.Add([]string{tableName, username, queryType}, duration)
 }
 
-func (qre *QueryExecutor) generateBindVarsForViewDDLInsert(createView *sqlparser.CreateView) map[string]*querypb.BindVariable {
+func generateBindVarsForViewDDLInsert(createView *sqlparser.CreateView) map[string]*querypb.BindVariable {
 	bindVars := make(map[string]*querypb.BindVariable)
 	bindVars["table_name"] = sqltypes.StringBindVariable(createView.ViewName.Name.String())
 	bindVars["view_definition"] = sqltypes.StringBindVariable(sqlparser.String(createView.Select))
 	bindVars["create_statement"] = sqltypes.StringBindVariable(sqlparser.String(createView))
 	return bindVars
 }
+
+func (qre *QueryExecutor) GetSchemaDef(tableType querypb.TableType, tableNames []string) (map[string]string, error) {
+	switch tableType {
+	case querypb.TableType_VIEWS:
+		return qre.getViewDef(tableNames)
+	case querypb.TableType_TABLES:
+		return nil, vterrors.Errorf(vtrpcpb.Code_INVALID_ARGUMENT, "invalid table type %v", tableType)
+	}
+	return nil, vterrors.Errorf(vtrpcpb.Code_INVALID_ARGUMENT, "invalid table type %v", tableType)
+}
+
+func (qre *QueryExecutor) getViewDef(tableNames []string) (map[string]string, error) {
```

Comment:
```suggestion
func (qre *QueryExecutor) getViewDefinitions(tableNames []string) (map[string]string, error) {
```

Reply:
shouldn't it be `viewNames` instead of `tableNames`?

Reply:
everything is a table only the type is different. `VIEW / BASE TABLE` i.e. how MySQL differentiates them.